evil-collection.el

```diff
@@ -175,6 +193,55 @@ Evil Collection for that mode. More arguments may be added in the future, so
 functions added to this hook should include a \"&rest _rest\" for forward
 compatibility.")
 
+(defun evil-collection-define-key (state package map-sym &rest bindings)
+  "Wrapper for `evil-define-key*' with additional features.
+Filter keys on the basis of `evil-collection-key-whitelist' and
+`evil-collection-key-blacklist'. Store bindings in
+`evil-collection-bindings-record'."
+  (declare (indent defun))
+  (let* ((whitelist (mapcar 'kbd evil-collection-key-whitelist))
+         (blacklist (mapcar 'kbd evil-collection-key-blacklist))
+         (record (cdr-safe (assoc package evil-collection-bindings-record))))
+    (while bindings
+      (let ((key (pop bindings))
+            (def (pop bindings)))
+        (when (or (and whitelist (member key whitelist))
+                  (not (member key blacklist)))
+          (push (cons key def) record)
+          (eval-after-load package
+            `(condition-case err
+                 (evil-define-key* ',state ,map-sym ,key ',def)
+               (error
+                (message "evil-collection: Bad binding %s"
+                         '(evil-define-key* ',state ,map-sym ,key ',def))))))))
+    (setq record (nreverse record))
+    (if (assoc package evil-collection-bindings-record)
+        (setcdr (assoc package evil-collection-bindings-record) record)
+      (push (cons package record) evil-collection-bindings-record))
```

Comment:
What about using a hashmap instead of an alist?  It would simplify this `if` as well as the code above a little bit.  I think it's also more readable.

Reply:
I can do that. I figured since I iterate over the list later, a list makes a little more sense, but a hashmap is fine too.